build/dockerfiles/rhel.Dockerfile

```diff
@@ -8,30 +8,28 @@
 # Contributors:
 #   Red Hat, Inc. - initial API and implementation
 #
-# this container build creates configbump binary
+# this container build creates configbump binary in a container
+# see also brew.Dockerfile
+
+# https://registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.7-1049.1675784874
```

Comment:
I'm more in favor of using a go-toolset builder image (can be helpful for verifying Go version used for the build, etc) and then copying built artifacts into a basic ubi8-minimal image, but if that's how it's done here then so be it.

Reply:
These dockerfiles could probably be templated off something like https://github.com/devfile/devworkspace-operator/blob/main/build/Dockerfile

Reply:
https://github.com/devfile/devworkspace-operator/blob/main/build/Dockerfile is not cachito-compliant. 

And I switched from using a two stage with go-toolset and then a ubi target to a single stage, as that's what we have been using downstream for some time. Could have gone the other way, but given rhel.Dockerfile isn't even being USED in the configbump build (because still using Alpine for some reason), figured that having upstream consistent with what's actually shipping downstream was better. 

Would you be +1 to switch the configbump release process to use ubi8 instead of alpine, so we can stop having 3 ways to build it?

Reply:
> helpful for verifying Go version used for the build

The build log will show which RPMs / modules are installed: 
```
golang-1.18.9-1.module+el8.7.0+17640+84246675.x86_64
golang-bin-1.18.9-1.module+el8.7.0+17640+84246675.x86_64
golang-src-1.18.9-1.module+el8.7.0+17640+84246675.noarch
```

Reply:
I'm +1 on removing alpine in general but it's not my decision to make -- I'm a bystander on this one.

As for the devworkspace operator Dockerfile -- the cachito-compliant version just replaces the builder `COPY` steps with
```dockerfile
COPY $REMOTE_SOURCE $REMOTE_SOURCE_DIR

WORKDIR ${REMOTE_SOURCE_DIR}/app/
```

The issue with Go version isn't that you don't know the exact version of Go, it's that Go can update minor versions freely, which can cause issues -- e.g. bumping to Go 1.17 to 1.18 will change your go.mod file.